### PR TITLE
⬆️ Bump yq version

### LINKED ISSUE #40: ⬆️  Bump yp binary version
### Feature Suggestion

- there is a new major release for the binary file

### Possible Implementation

- first: use the new version + adapt the command calls ([migration guide](https://mikefarah.gitbook.io/yq/upgrading-from-v3))
- second: using the GitHub action so it is not necessary to download the binary

### PR DESCRIPTION
**Pull request by flaxel**
### Type of Change

- [ ] Bugfix
- [x] Enhancement / new feature
- [ ] Refactoring
- [ ] Documentation

### Description

- bump yq version
- using of the yq action is not necessary because it is already part of the ubuntu image
- test it with [this pr](https://github.com/Staffbase/backstage-app/pull/113) in [this run](https://github.com/Staffbase/backstage-app/actions/runs/2601237628)

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [x] Review the [Contributing Guideline](../blob/master/CONTRIBUTING.md) and sign CLA
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -151,7 +151,7 @@ runs:
         }
 
         # download and install yq
-        sudo wget -O /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
+        sudo wget -O /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/v4.25.3/yq_linux_amd64
```
Comment (flaxel) [inline action.yml:154]: download the new version

File: action.yml [hunk 2]
```diff
@@ -163,40 +163,44 @@ runs:
         if [[ ( $GITHUB_REF == refs/heads/master || $GITHUB_REF == refs/heads/main ) && -n "${{ inputs.gitopsstage }}" ]]; then
           echo "Run update for STAGE"
           while IFS= read -r line; do
+            array=($line)
```
Comment (flaxel) [inline action.yml:166]: split input by blank

File: action.yml [hunk 3]
```diff
@@ -163,40 +163,44 @@ runs:
...
             echo "Check if path $line exists and get old current version"
-            yq r -e $line
+            yq -e .${array[1]} ${array[0]}
             echo "Run update $line ${{ inputs.dockerregistry }}/${{ inputs.dockerimage }}:${{ steps.preparation.outputs.tag }}"
-            yq w -i $line ${{ inputs.dockerregistry }}/${{ inputs.dockerimage }}:${{ steps.preparation.outputs.tag }}
+            yq -i .${array[1]}=\"${{ inputs.dockerregistry }}/${{ inputs.dockerimage }}:${{ steps.preparation.outputs.tag }}\" ${array[0]}
```
Comment (flaxel) [inline action.yml:170]: migrated the calls from version 3 to 4

File: action.yml [hunk 4]
```diff
@@ -150,53 +159,50 @@ runs:
           fi
         }
 
-        # download and install yq
-        sudo wget -O /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
-        sudo chmod +x /usr/local/bin/yq
```
Comment (flaxel) [inline action.yml:155]: yq is already part of the github action with ubuntu 20: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#tools

File: action.yml [hunk 5]
```diff
@@ -150,53 +159,50 @@ runs:
...
-
+        # configure git user
         git config --global user.email "${{ inputs.gitopsemail }}" && git config --global user.name "${{ inputs.gitopsuser }}"
 
-        git clone https://${{ inputs.gitopsuser }}:${{ inputs.gitopstoken }}@github.com/${{ inputs.gitopsorganization }}/${{ inputs.gitopsrepository }}.git .github/${{ inputs.gitopsrepository }}
-        cd .github/${{ inputs.gitopsrepository }}
-        echo "Repository ${{ inputs.gitopsorganization }}/${{ inputs.gitopsrepository }} was cloned successfully "
```
Comment (flaxel) [inline action.yml:161]: Use explicit checkout action

**Review by 0x46616c6b — APPROVED** [review]

**Pull request merged**